lib/src/util/context.dart

```diff
@@ -129,68 +130,62 @@ class Context<TValue> {
   final UiFactory<ConsumerProps<TValue>> Consumer;
 }
 
-/// [ProviderProps] is a typed props class for the [Context.Provider] from a [Context] object created with [createContext].
+// Trigger generation of concrete props classes we can use below in _ProviderPropsImpl/ _ConsumerPropsImpl.
+UiFactory<ProviderProps> _Provider = _$_Provider; // ignore: unused_element
+UiFactory<ConsumerProps> _Consumer = _$_Consumer; // ignore: unused_element
+
+/// A concrete implementation of [ProviderProps], which, unlike a [UiFactory],
+/// can be used to instantiate it with a generic parameter.
+///
+/// We rely on this behavior within [Context].
+///
+/// Uses the generated concrete props class so we don't have to manually implement generated members
+/// and apply generated `$`-prefixed props mixin class.
+// ignore: deprecated_member_use_from_same_package
+typedef _ProviderPropsImpl<TValue> = _$$ProviderProps<TValue>;
+
+/// A concrete implementation of [ConsumerProps], which, unlike a [UiFactory],
+///
+/// We rely on this behavior within [Context].
+///
+/// Uses the generated concrete props class so we don't have to manually implement generated members
+/// and apply generated `$`-prefixed props mixin class.
+// ignore: deprecated_member_use_from_same_package
+typedef _ConsumerPropsImpl<TValue> = _$$ConsumerProps<TValue>;
+
+/// A typed props class for the [Context.Provider] from a [Context] object created with [createContext].
 ///
 /// Props:
 ///
-/// * [value] The value that you want to provide to all consumers.
+/// * [_ProviderPropsMixin.value] The value that you want to provide to all consumers.
 ///
 /// See: <https://react.dev/reference/react/createContext#provider>
-class ProviderProps<TValue> extends builder_helpers.UiProps {
-  ProviderProps([Map? backingMap]) : this.props = backingMap ?? JsBackedMap();
-
-  @override
-  final Map props;
-
-  @override
-  String get propKeyNamespace => '';
-
-  @override
-  bool get $isClassGenerated => true;
-
-  @override
-  String $getPropKey(accessMap) => _getPropKey(accessMap, (backingMap) => ProviderProps(backingMap));
+class ProviderProps<TValue> = UiProps with _ProviderPropsMixin<TValue>;
 
-  TValue? get value => props['value'] as TValue?;
-  set value(TValue? v) => props['value'] = v;
+// Private to mirror `ConsumerPropsMixin` being private.
+@Props(keyNamespace: '')
+mixin _ProviderPropsMixin<TValue> on UiProps {
+  late TValue value;
```

Comment:
This probably deserves a doc comment.

Reply:
Addressed in [2574c34](https://github.com/Workiva/over_react/pull/876/commits/2574c344f4a69996004b4c688ae10625bd928939)